test/utils/asserts.cpp

```diff
@@ -20,4 +20,19 @@ std::ostream& operator<<(std::ostream& os, ExecutionResult result)
     os.flags(format_flags);
     return os;
 }
+
+namespace test
+{
+std::ostream& operator<<(std::ostream& os, const TypedExecutionResult& result)
+{
+    const auto format_flags = os.flags();
+
+    os << ExecutionResult{result};
+
+    os << " (type " << std::hex << static_cast<int>(result.type) << ")";
```

Comment:
`operator<<` for `ValType` would be nice

Reply:
Not needed any more.